crates/biome_cli/src/commands/format.rs

```diff
@@ -157,12 +159,22 @@ pub(crate) fn format(
     let (vcs_base_path, gitignore_matches) =
         configuration.retrieve_gitignore_matches(&session.app.fs, vcs_base_path.as_deref())?;
 
-    if since.is_some() && !changed {
-        return Err(CliDiagnostic::incompatible_arguments("since", "changed"));
+    if since.is_some() {
+        if !changed {
+            return Err(CliDiagnostic::incompatible_arguments("since", "changed"));
+        }
+        if staged {
+            return Err(CliDiagnostic::incompatible_arguments("since", "staged"));
+        }
     }
 
     if changed {
+        if staged {
+            return Err(CliDiagnostic::incompatible_arguments("changed", "staged"));
+        }
         paths = get_changed_files(&session.app.fs, &configuration, since)?;
+    } else if staged {
+        paths = get_staged_files(&session.app.fs)?;
     }
```

Comment:
Considering how the code is repeated, maybe it makes more sense to have a function in `commands/mod.rs` that does the check, and each command imports it and calls it.

Reply:
Extracted 🏗️ .